brave/brave-core
# Adds Brave Ads launch notification for existing Rewards users

Fixes: https://github.com/brave/brave-browser/issues/2914
UI PR: https://github.com/brave/brave-ui/pull/346

Note: This is only intended to be shown to existing users of Rewards who have not enabled Ads yet

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
1. Use a `<0.63` version of Brave to create a profile.
2. Using the older version, simply enable rewards and do nothing else
3. Create a copy of the profile appropriately titled `Brave-Browser-Development`
4. Build version from this PR (passing `--brave-ads-debug` as a flag, this reduces the time to delete notification from 7 days to 5 mins) and start Brave
5. Navigate to the panel and confirm that the Ads notification prompting the user to turn on ads is available
6. Confirm that click "Turn on ads" takes you to `brave://rewards` and dismisses the notification.
7. Restart browser
8. Confirm that the notification is not available

Run through the above steps again, except after step 2, disable Rewards before copying the profile. Ensure that when upgrading the notification is not shown.

Also, test this with a clean profile. Opt-in to Rewards, confirm that you don't get the notification (opting in enables ads)

## Reviewer Checklist:


cc: @mandar-brave @jsecretan
